src/core/table/table.jsx

```diff
@@ -0,0 +1,268 @@
+import React, { useState, useMemo } from 'react';
+import PropTypes from 'prop-types'
+import _ from 'lodash';
+
+import Box from '@mui/material/Box';
+import {
+    Table as MuiTable,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableHead,
+    TablePagination,
+    TableRow,
+    TableSortLabel,
+    Paper,
+    Checkbox,
+    Radio,
+    CircularProgress
+} from '@mui/material';
+import { visuallyHidden } from '@mui/utils';
+
+/*
+    @columns: Array of objects
+    @column: {
+        id: Unique identifier
+        accessor: Property to be accessed
+        label: Header label
+        template: custom template
+    }
+*/
+
+export const Table = ({
+    data,
+    columns = [],
+    loading = false,
+    onRowClick = _.noop,
+    onRowSelect = _.noop,
+    defaultOrder = 'asc',
+    defaultOrderBy = '',
+    multiSelect = false
+}) => {
+    const [order, setOrder] = useState(defaultOrder);
+    const [orderBy, setOrderBy] = useState(defaultOrderBy);
+    const [selected, setSelected] = React.useState([]);
+    const [page, setPage] = React.useState(0);
+    const [rowsPerPage, setRowsPerPage] = React.useState(5);
+
+    const processedData = useMemo(() => {
+        return data.map(entry => {
+            return {
+                original: entry,
+                _id: _.uniqueId('table__entry__')
+            };
+        });
+    }, [data]);
+
+    const processedColumns = useMemo(() => columns.filter(({ show = true }) => show));
+
+    const handleRequestSort = (event, property) => {
+        const isAsc = orderBy === property && order === 'asc';
+        setOrder(isAsc ? 'desc' : 'asc');
+        setOrderBy(property);
+    };
+
+    const handleClick = (event, rawInfo) => {
+        const selectedIndex = selected.findIndex(entry => entry._id === rawInfo._id);
+        let newSelected = [];
+
+        if (selectedIndex === -1) {
+            newSelected = newSelected.concat(selected, rawInfo);
+        } else if (selectedIndex === 0) {
+            newSelected = newSelected.concat(selected.slice(1));
+        } else if (selectedIndex === selected.length - 1) {
+            newSelected = newSelected.concat(selected.slice(0, -1));
+        } else if (selectedIndex > 0) {
+            newSelected = newSelected.concat(
+                selected.slice(0, selectedIndex),
+                selected.slice(selectedIndex + 1)
+            );
+        }
+
+        if (!multiSelect) {
+            newSelected = [rawInfo];
+        }
+
+        setSelected(newSelected);
+        onRowSelect(newSelected);
+        onRowClick(rawInfo);
+    };
+
+    const createSortHandler = (property) => (event) => {
+        handleRequestSort(event, property);
+    };
+
+    const isSelected = (id) => selected.findIndex((entry) => entry._id === id) !== -1;
+
+    const emptyRows = page > 0 ? Math.max(0, (1 + page) * rowsPerPage - processedData.length) : 0;
```

Comment:
The Math.max is already going establish a lower limit to `0`, no?

```suggestion
    const emptyRows = Math.max(0, (1 + page) * rowsPerPage - processedData.length);
```